src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -0,0 +1,166 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                    TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <algorithm>
+
+#include <rpp/operators/fwd/combine_latest.hpp>
+#include <rpp/subscribers/constraints.hpp>
+#include <rpp/sources/create.hpp>
+#include <rpp/utils/functors.hpp>
+
+#include <rpp/operators/details/subscriber_with_state.hpp> // create_subscriber_with_state
+
+
+IMPLEMENTATION_FILE(combine_latest_tag);
+
+namespace rpp::details
+{
+
+/**
+ * \brief "combine_latest" operator (an OperatorFn used by "lift").
+ */
+template<constraint::decayed_type Type, typename TCombiner, constraint::observable ...TOtherObservable>
+struct combine_latest_impl
+{
+    TCombiner m_combiner;
+    std::tuple<TOtherObservable...> m_other_observables;
+
+private:
+    static constexpr size_t INDEX_OF_SOURCE_TYPE = 0;
+
+    using DownstreamType = utils::decayed_invoke_result_t<TCombiner, Type, utils::extract_observable_type_t<TOtherObservable>...>;
+
+    /**
+     * \brief The coordinator that caches the values from all the observables and dispatches latest caches to the observer. Note the emission is only sent to the observer when all the observables at least emits once.
+     */
+    template<constraint::decayed_type SourceValueType, constraint::decayed_type ...OtherValueTypes>
+    struct combine_latest_coordinator_t
+    {
+        explicit combine_latest_coordinator_t(TCombiner combiner) : m_combiner(combiner) {}
```

Comment:
```suggestion
        explicit combine_latest_coordinator_t(const TCombiner& combiner) : m_combiner(combiner) {}
```

Reply:
Oh this is ok bc the `combine_latest_impl` lives longer than `combine_latest_coordinator_t`?